include/zim/search.h

```diff
@@ -217,6 +217,30 @@ class Search
   friend class Searcher;
 };
 
+class SuggestionSearch
+{
+    public:
+        SuggestionSearch(SuggestionSearch&& s);
+        SuggestionSearch& operator=(SuggestionSearch&& s);
+        ~SuggestionSearch();
+
+        const SuggestionResultSet getResults(int start, int end) const;
+
+        int getEstimatedMatches() const;
+
+    private: // methods
+        SuggestionSearch(std::shared_ptr<InternalDataBase> p_internalDb, const Query& query);
+        SuggestionSearch(const Query& query);
```

Comment:
By definition a `Search` or a `SuggestionSearch` is a wrapper around a "database" (xapian or archive) and a `Query`. It is "nonsense" to create a `SuggestionSearch` on a `Query` only.

And it is dangerous to do that. The `SuggestionSearcher` is returning a wrong `SuggestionSearch` without `mp_internalDb` but user could still use methods on it and they never check `mp_internalDb` is not null.